### PR TITLE
[ceres] update to v2.1.0

### LINKED ISSUE #23818: [ceres] update to 2.1.0
Library name: ceres (Ceres Solver) http://ceres-solver.org/

New version number: 2.1.0

Other information that may be useful (release notes, etc...)

Their CMake build files have been cleaned and modernised, so there are likely changes needed to the vcpkg-specific patches (or they might not be required anymore). See the change log http://ceres-solver.org/version_history.html#id1.

@fran6co I think you were leading the efforts on the ceres update to 2.0 in vcpkg (thanks a lot for that!), so I thought it's a good idea to ping you.

### PR DESCRIPTION
**Pull request by Adela0814**
Fix #23818
update to latest version v2.1.0

Update patch to fix dependences not find issue, 
Add feature `CUDA`

All features have been tested successfullly in the following triplet:

x64-windows
x86-windows
x64-windows-static

### CODE REVIEW

File: ports/ceres/0003_fix_exported_ceres_config.patch [hunk 1]
```diff
@@ -31,12 +31,36 @@ index 4745483..4c461e8 100644
...
+-      ceres_message(STATUS "Found required Ceres dependency: gflags")
+-    else()
+-      ceres_report_not_found("Missing required Ceres "
+-        "dependency: gflags (not found, or not found as exported CMake target).")
+-    endif()
+-  endif()
+-endif(CERES_USES_MINIGLOG)
++# SuiteSparse
++if (@SUITESPARSE@)
++    find_dependency(suitesparse CONFIG)
```
Comment (dg0yt) [inline ports/ceres/0003_fix_exported_ceres_config.patch:62]: For consistency with `CMakeLists.txt`, I would prefer `SuiteSparse` spelling here (and `CONFIG` there). Case matters in some cases.

**Comment by strega-nil-ms** [comment]
Thanks!

**Comment by cenit** [comment]
hash is wrong? why it didn't trigger any problem here before? caching?

File: ports/ceres/vcpkg.json [hunk 2]
```diff
@@ -17,6 +17,13 @@
     }
   ],
   "features": {
+    "cuda": {
+      "description": "Build darknet with support for CUDA",
```
Comment (cenit) [inline ports/ceres/vcpkg.json:21]: darknet? :)

File: ports/ceres/find-package-required.patch [hunk 3]
```diff
@@ -1,58 +1,112 @@
...
  
    # Check for SuiteSparse and dependencies.
--  find_package(SuiteSparse)
-+  find_package(SuiteSparse REQUIRED)
-   if (SUITESPARSE_FOUND)
-     # On Ubuntu the system install of SuiteSparse (v3.4.0) up to at least
-     # Ubuntu 13.10 cannot be used to link shared libraries.
-@@ -300,7 +300,7 @@ endif (SUITESPARSE)
+-  find_package(SuiteSparse 4.0 COMPONENTS CHOLMOD SPQR)
++  find_package(suitesparse CONFIG REQUIRED)
```
Comment (cenit) [inline ports/ceres/find-package-required.patch:19]: is letter case ok?

Comment (Adela0814) [reply ports/ceres/find-package-required.patch:19]: yes, I checked the usage of `suitesparse` port, is letter case, so I changed this to match.

File: ports/ceres/vcpkg.json (hunk 2 above)
Comment (Adela0814) [reply ports/ceres/vcpkg.json:21]: sorry, I will change it.

**Pull request merged**